gateway/webhook/webhookTransformer.go

```diff
@@ -10,11 +10,92 @@ import (
 	"strings"
 	"time"
 
+	"github.com/rudderlabs/rudder-go-kit/config"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
 	"github.com/rudderlabs/rudder-server/gateway/response"
+	"github.com/rudderlabs/rudder-server/services/transformer"
 	"github.com/rudderlabs/rudder-server/utils/httputil"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type sourceTransformAdapter interface {
+	getTransformerEvent(sourceId string, body []byte) ([]byte, error)
+	getTransformerURL(sourceType string) string
+}
+
+type v0Adapter struct {
+	version string
+	bt      *batchWebhookTransformerT
```

Comment:
We shouldn't leak this whole dependency here just for getting a source for a sourceID (1 single resolver method).
Even better, we shouldn't require a resolver method at all